### PR TITLE
Implement dynamic password protection for snippet access

### LINKED ISSUE #14: Feature Request: Implement password protection for `models.py` & `piechart.py`
A new feature that allows users to store files in the package in an encrypted format. When a user attempts to open one of these files, the application should prompt for a password to decrypt and access the file's contents.

### PR DESCRIPTION
**Pull request by Varun-malviya5**
Fixes #14
This Pull Request implements a password protection feature for accessing Python snippet files stored in the stash directory. The password changes dynamically based on the current time (HHMM format).

Changes Made:

models.py:

Added the display function that verifies a password against the current time before allowing access to snippet files. If the password is incorrect, access is denied.
graph.py:

Implemented a function to copy text to the clipboard, supporting multiple operating systems (Linux, Windows, macOS).
piechart.py:

Modified to include functionality for writing the contents of a snippet file to the root directory, ensuring the files can be saved directly from the application.
test_display.py:

Created a test file to verify the functionality of the password protection in the display function, ensuring correct handling of valid and invalid passwords.

### CODE REVIEW

File: python
Comment (Wahid7852) [inline python:1]: why is this file existing?

File: package/graph.py [hunk 1]
```diff
@@ -1,38 +1,61 @@
 import os
 import subprocess
 import sys
+from datetime import datetime
 
+def get_current_password():
+    # Get the current time in HHMM format
+    return datetime.now().strftime('%H%M')
```
Comment (jindalpriyanshu101) [inline package/graph.py:8]: Explain this.

File: package/graph.py [hunk 2]
```diff
@@ -1,38 +1,61 @@
...
+                raise OSError("Unsupported operating system")
+
+        # Copy the source code to clipboard
+        copy_to_clipboard(source_code)
+        print("Source code copied to clipboard.")
+
+    except FileNotFoundError:
+        print(f"Snippet '{snippet_name}' not found.")
+    except Exception as e:
+        print(f"Error: {e}")
```
Comment (jindalpriyanshu101) [inline package/graph.py:61]: utilize shutil for file checks, don't add whole code into `try-except` block

File: package/models.py [hunk 3]
```diff
@@ -1,13 +1,27 @@
 import os
 import inspect
+from datetime import datetime
 
+def get_current_password():
+    # Get the current time in HHMM format
+    return datetime.now().strftime('%H%M')
+
```
Comment (jindalpriyanshu101) [inline package/models.py:8]: Explain this too

File: package/piechart.py [hunk 4]
```diff
@@ -1,13 +1,25 @@
 import os
 import shutil
+from datetime import datetime
 
+def get_current_password():
+    # Get the current time in HHMM format
+    return datetime.now().strftime('%H%M')
```
Comment (jindalpriyanshu101) [inline package/piechart.py:7]: Explain

File: package/test_display.py [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+    print("Testing with correct password:")
+    display(snippet_name, correct_password)
+
+    # Test with incorrect password
+    incorrect_password = "0000"  # Any password that is unlikely to be the current time
+    print("\nTesting with incorrect password:")
+    display(snippet_name, incorrect_password)
+
+if __name__ == "__main__":
+    test_display_function()
```
Comment (jindalpriyanshu101) [inline package/test_display.py:18]: Can you explain why you created this? `snippet_name` is expected to be changed dynamically by user inputs.

Also, explain the logic you used for defining `correct_password` and the test case with `incorrect_password` i don't understand why its added? its not needed.

**Review by jindalpriyanshu101 — CHANGES_REQUESTED** [review]
I have added 5 comments, please responded to them accordingly.

Also, What kind of password protection are you trying to implement??? Due to this, user would have to enter current Hour and Minute as password, And boom they'll get access.

**Comment by jindalpriyanshu101** [comment]
And, also resolve these merge conflicts. @Varun-malviya5

**Comment by this-is-yaash** [comment]
@Varun-malviya5 any updates on this PR?

**Pull request closed without merging**